### ISSUE TITLE
corrupted tarball data during clean install

### ISSUE THREAD

**Issue by pkolomytsev** [nvuillam/npm-groovy-lint#431]
Versions:
- python v3.8.19
- pre-commit v3.5.0
- npm-groovy-lint v15.0.2

Log:

```
$ pre-commit run -a
[INFO] Initializing environment for https://github.com/nvuillam/npm-groovy-lint.
[INFO] Installing environment for https://github.com/nvuillam/npm-groovy-lint.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
An unexpected error has occurred: CalledProcessError: command: ('/home/pkolomytsev/.cache/pre-commit/repov47v0tmo/node_env-default/bin/node', '/home/pkolomytsev/.cache/pre-commit/repov47v0tmo/node_env-default/bin/npm', 'pack')
return code: 254
stdout: (none)
stderr:
    (node:261589) ExperimentalWarning: Support for loading ES Module in require() is an experimental feature and might change at any time
    (Use `node --trace-warnings ...` to show where the warning was created)
    npm warn tarball tarball data for file:/home/pkolomytsev/.cache/pre-commit/repov47v0tmo/ (null) seems to be corrupted. Trying again.
    npm warn tarball tarball data for file:/home/pkolomytsev/.cache/pre-commit/repov47v0tmo/ (null) seems to be corrupted. Trying again.
    npm error code ENOENT
    npm error syscall lstat
    npm error path /home/pkolomytsev/.cache/pre-commit/repov47v0tmo/ib/example/Jenkinsfile
    npm error errno -2
    npm error enoent ENOENT: no such file or directory, lstat '/home/pkolomytsev/.cache/pre-commit/repov47v0tmo/ib/example/Jenkinsfile'
    npm error enoent This is related to npm not being able to find a file.
    npm error enoent
    npm error A complete log of this run can be found in: /home/pkolomytsev/.npm/_logs/2024-10-24T10_53_21_971Z-debug-0.log
```

I think the problem appeared after Node.js was updated (v23.0.0). pre-commit does something like this:

```
nodeenv node_env-default
source node_env-default/bin/activate
npm pack
```

These steps return the same result as above. If you downgrade the Node.js version, there will be no error:

```
nodeenv node_env-default -n 22.10.0
source node_env-default/bin/activate
npm pack
```

So a possible workaround is to create a file `$HOME/.nodeenv`:

```
[nodeenv]
node = 22.10.0
```

**UPD**:

Setting `language_version` also works:

```
repos:
  - repo: https://github.com/nvuillam/npm-groovy-lint
    rev: v15.0.0
    hooks:
      - id: npm-groovy-lint
        language_version: 22.10.0
        args: [--config=recommended-jenkinsfile]
      - id: format-npm-groovy-lint
        language_version: 22.10.0
        args: [--config=recommended-jenkinsfile]
```

**Comment by nvuillam** [comment]
It seems to be a Node23 + pre-commit problem , not a npm-groovy-lint , no ? :/

**Comment by pkolomytsev** [comment]
Yes, it was a Node.js bug (https://github.com/nodejs/node/issues/55410). v23.1.0 has already been released and contains a fix.
